### PR TITLE
test: empty or corrupt docs

### PR DESCRIPTION
**Pull request by cristianmtr**
- No content
- [x] Only tags
- [x] Only embedding
- [x] Different mime types
- [x] Malformed content (assign to text and assing mime-type image. Test in parallel pods flow)
- [x] Different shapes, dimensionality

### CODE REVIEW

File: jina/executors/indexers/vector.py [hunk 1]
```diff
@@ -149,9 +149,11 @@ def _delete(self, keys):
                 self._size -= 1
 
     def delete(self, keys: Sequence[int], *args, **kwargs) -> None:
-        if kwargs.get('keys_precomputed') is not True:
-            keys = self._filter_nonexistent_keys(keys, self.ext2int_id.keys(), self.save_abspath)
-        self._delete(keys)
+        # if we dont check here, it crashes on `self.ext2int_id.keys()`
+        if self.size > 0:
+            if kwargs.get('keys_precomputed') is not True:
```
Comment (JoanFM) [inline jina/executors/indexers/vector.py:154]: please, use private `_delete` method and do not read from `kwargs` as it is done here, I thought this was removed from the code

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/executors/indexers/vector.py [hunk 2]
```diff
@@ -149,9 +151,7 @@ def _delete(self, keys):
                 self._size -= 1
 
     def delete(self, keys: Sequence[int], *args, **kwargs) -> None:
-        if kwargs.get('keys_precomputed') is not True:
-            keys = self._filter_nonexistent_keys(keys, self.ext2int_id.keys(), self.save_abspath)
-        self._delete(keys)
+        self._delete(keys, keys_precomputed=False)
```
Comment (cristianmtr) [inline jina/executors/indexers/vector.py:154]: @JoanFM Is this what you mean?

Comment (JoanFM) [reply jina/executors/indexers/vector.py:154]: yes

File: tests/integration/crud_corrupted/test_crud_corrupted.py [hunk 3]
```diff
@@ -93,21 +94,25 @@ def validate_results(resp):
 EMBEDDING_SHAPE = (7)
 
 
-def random_docs_only_embedding(nr_docs, start=0):
+def random_docs_only_embedding(nr_docs, mime_type='plain/text', start=0):
     for i in range(start, nr_docs + start):
         d = Document()
         d.embedding = np.random.random(EMBEDDING_SHAPE)
+        d.mime_type = mime_type
```
[the remark this answers is not in the document]
Comment (cristianmtr) [reply tests/integration/crud_corrupted/test_crud_corrupted.py]: You mean the `content`, or the `embedding`?

[the remark this answers is not in the document]
Comment (JoanFM) [reply tests/integration/crud_corrupted/test_crud_corrupted.py]: the content yes

**Review by hanxiao — APPROVED** [review]

**Pull request merged**